include/tvm/ir/type.h

```diff
@@ -181,6 +181,55 @@ class PointerType : public Type {
   TVM_DEFINE_OBJECT_REF_METHODS(PointerType, Type, PointerTypeNode);
 };
 
+/*!
+ * \brief Low-level texture type.
+ *
+ *  TextureType represents type hints in the TIR to be
+ *  passed to the final code generator.
+ *
+ *  TextureType should not occur in the high-level analysis.
+ *
+ * \sa TextureType
+ */
+class TextureTypeNode : public TypeNode {
+ public:
+  /*!
+   * \brief The base type of the texture.
+   */
```

Comment:
Given this is quite close to PointerType, perhaps we should think about instead introducing a scope flag (str for now) to the PointerType, instead of introducing a new type

Reply:
@tqchen @csullivan , just curious (and for my understanding) why cant we point the Var to a buffer (using the buffer map) and buffer has a way to mention the scope ?

Came here by looking at why this is needed in #8017

Reply:
A storage specific type or scope is helpful in the IR so it can be tracked throughout various transformations including during [codegen](https://github.com/apache/tvm/pull/7686/files#diff-ee4125c7c00371317aff30351177693f0449409f24477f5f85cba7110d988616R216).